R/maraca.R

```diff
@@ -610,202 +684,101 @@ plot.hce <- function(x, continuous_grid_spacing_x = 10, trans = "identity",
   n <- dplyr::n
   `%>%` <- dplyr::`%>%`
 
-  meta1 <- HCE %>%
+  metaMissing <- HCE %>%
+    dplyr::group_by(outcome) %>%
+    dplyr::summarise(
+      missing = sum(is.na(value))
+    )
+
+  meta1 <- HCE  %>%
+    dplyr::filter(!is.na(value)) %>%
     dplyr::group_by(outcome) %>%
     dplyr::summarise(
       n = n(),
       proportion = n / dim(HCE)[1] * 100,
-      maxday = max(value, na.rm = TRUE)) %>%
+      maxday = max(value, na.rm = TRUE)
+    ) %>%
     dplyr::mutate(
       startx = c(0, cumsum(utils::head(proportion, -1))),
       endx = cumsum(proportion),
       starty = 0,
       n.groups = length(unique(outcome))
     )
 
-  meta2 <- HCE %>%
+  meta2 <- HCE  %>%
+    dplyr::filter(!is.na(value)) %>%
     dplyr::group_by(outcome, arm) %>%
     dplyr::summarise(n = n(), proportion = n / dim(HCE)[1] * 100) %>%
     tidyr::pivot_wider(names_from = arm, values_from = c(n, proportion))
 
   meta <- dplyr::left_join(meta1, meta2, "outcome")
+  meta <- dplyr::left_join(meta, metaMissing, "outcome")
 
   return(meta)
 }
 
-# Performs the survmod calculation
-.compute_survmod_by_outcome <- function(
-    HCE, meta, tte_outcomes, continuous_outcome, arm_levels,
-    fixed_followup_days
-  ) {
+# Calculates the cumulative distribution for TTE outcomes
+.compute_ecdf_by_outcome <- function(
+  HCE, meta, tte_outcomes, continuous_outcome, arm_levels,
+  fixed_followup_days
+) {
 
-  endpoints <- c(tte_outcomes, continuous_outcome)
-  vars <- dplyr::vars
   `%>%` <- dplyr::`%>%`
+  n <- dplyr::n
+  .data <- rlang::.data
 
-  Surv <- survival::Surv # nolint
-
-  censored_continuous_fits <- list()
-  for (i in seq_along(tte_outcomes)) {
-    HCE_focused <- .hce_survival_focus(
-      HCE, i, tte_outcomes, fixed_followup_days)
-
-    censored_continuous_fit <- survival::survfit(
-      Surv(time = kmday, event = (outcome != continuous_outcome)) ~ arm,
-      data = HCE_focused
-    )
-
-    censored_continuous_fits[[i]] <- censored_continuous_fit
-
-    # Create survival model dataset
-    survmod_data_row <- cbind(
-      ggplot2::fortify(censored_continuous_fit),
-      outcome = tte_outcomes[[i]]
-    )
-
-    n <- dplyr::n
-    # remove first and last point
-    survmod_data_row <- survmod_data_row
-
-    if (i == 1) {
-      survmod_data_row <- survmod_data_row %>%
-        dplyr::group_by(strata) %>%
-        dplyr::slice(1:(n() - 1))
-    } else if (i == length(tte_outcomes)) {
-      survmod_data_row <- survmod_data_row %>%
-        dplyr::group_by(strata) %>%
-        dplyr::slice(2:(n() - 1))
-    } else {
-      survmod_data_row <- survmod_data_row %>%
-        dplyr::group_by(strata) %>%
-        dplyr::slice(2:(n() - 1))
-    }
+  num_tte_outcomes <- length(tte_outcomes)
+  HCE$t_cdf <- (num_tte_outcomes + 2) * fixed_followup_days
 
-    if (i == 1) {
-      survmod_data <- survmod_data_row
-    } else {
-      survmod_data <- rbind(
-        survmod_data,
-        survmod_data_row
-      )
-    }
+  for (i in 1:num_tte_outcomes) {
```

Comment:
never use 1:val to iterate. Always seq_length or seq_along.

Reply:
Changed